corehq/apps/es/tests/test_elastic_sync_multiplexed_command.py

```diff
@@ -0,0 +1,152 @@
+            manager.index_create(f'test_{self.index}')
+            manager.index_put_mapping(f'test_{self.index}', self.type, self.mapping)
+        # manager.index_put_alias(f'test_{self.index}', self.alias)
+        # es = get_client()
+        # initialize_index_and_mapping(es, self)
+        if not manager.index_exists(f'test_{self.secondary_index}'):
+            manager.index_create(f'test_{self.secondary_index}')
+            manager.index_put_mapping(f'test_{self.secondary_index}', self.type, self.mapping)
+
+    def remove_index(self):
+        manager.index_delete(f'test_{self.index}')
+        manager.index_delete(f'test_{self.secondary_index}')
+
+
+class TestElasticSyncMultiplexedCommand(SimpleTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.adapter = create_document_adapter(
+            TestDocumentAdapter,
+            ReIndexTestHelper.index,
+            ReIndexTestHelper.type,
+            secondary=ReIndexTestHelper.secondary_index
+        )
+
+        _DOC_ADAPTERS_BY_INDEX[ReIndexTestHelper.index] = cls.adapter
+        cls.es = get_client()
+        register(ReIndexTestHelper, ReIndexTestHelper.cname)
+        # registering another test index
+        register(TEST_INDEX_INFO, TEST_INDEX_INFO.hq_index_name)
+        cls.util = ESSyncUtil()
+
+    @classmethod
+    def tearDownClass(cls):
+        _DOC_ADAPTERS_BY_INDEX.pop(ReIndexTestHelper.index)
+        deregister(ReIndexTestHelper.cname)
+        deregister(TEST_INDEX_INFO)
+        super().tearDownClass()
+
+    def setUp(self):
+        super().setUp()
+        ReIndexTestHelper().setup_indexes()
+
+    def tearDown(self):
+        ReIndexTestHelper().remove_index()
+        return super().tearDown()
+
+    def test_invalid_index_canonical_raises(self):
+        with self.assertRaises(CommandError):
+            call_command(COMMAND_NAME, 'start', 'random_alias')
+
+    def test_not_mutliplexed_index_raises(self):
+        with self.assertRaises(IndexNotMultiplexedException):
+            call_command(COMMAND_NAME, 'start', TEST_INDEX_INFO.hq_index_name)
+
+    @patch('corehq.apps.es.management.commands.elastic_sync_multiplexed.check_task_progress')
+    @patch('corehq.apps.es.management.commands.elastic_sync_multiplexed.ESSyncUtil.perform_cleanup')
+    @patch('corehq.apps.es.management.commands.elastic_sync_multiplexed.ESSyncUtil.start_reindex_in_es')
+    def test_pass_multiplexed_index_raise_no_errors(self, sync_mock, cleanup_mock, _):
+        sync_mock.return_value = {
+            "task": "task_key"
+        }
+        call_command(COMMAND_NAME, 'start', 'cases')
+
+    def test_reindex_command_copies_all_documents(self):
+        self.adapter.primary.index(TestDoc('key_2', 'val'))
+        self.adapter.primary.index(TestDoc('key', 'value'))
+
+        call_command(COMMAND_NAME, 'start', ReIndexTestHelper.cname)
+        self.es.indices.refresh(self.adapter.secondary.index_name)
+
+        self.assertEqual(
```

Comment:
It would be nice to test that the IDs match.

Reply:
https://github.com/dimagi/commcare-hq/pull/32370/commits/774c71f1101b5c197f75b685e46ad874083ad43b